packages/wdio-browserstack-service/src/request-handler.ts

```diff
@@ -1,89 +1,116 @@
     public pendingUploads = 0
+    private readonly callback: Function|undefined
     public static tearDownInvoked = false
 
     static instance: RequestQueueHandler
 
     // making it private to use singleton pattern
-    private constructor() {}
+    private constructor(callback: Function) {
+        this.callback = callback
+        this.startEventBatchPolling()
+    }
 
-    public static getInstance(): RequestQueueHandler {
-        if (!RequestQueueHandler.instance) {
-            RequestQueueHandler.instance = new RequestQueueHandler()
+    public static getInstance(callback?: Function): RequestQueueHandler {
+        if (!RequestQueueHandler.instance && callback) {
+            RequestQueueHandler.instance = new RequestQueueHandler(callback)
         }
         return RequestQueueHandler.instance
     }
 
-    start () {
-        if (!this.started) {
-            this.started = true
-            this.startEventBatchPolling()
-        }
-    }
+    // add (event: UploadType) {
+    //     if (!process.env.BS_TESTOPS_BUILD_COMPLETED) {
+    //         return {
+    //             proceed: false
+    //         }
+    //     }
+    //
+    //     // if (!BATCH_EVENT_TYPES.includes(event.event_type)) {
+    //     //     return {
+    //     //         proceed: true
+    //     //     }
+    //     // }
+    //
+    //     // if (event.logs && event.logs[0] && event.logs[0].kind === 'TEST_SCREENSHOT') {
+    //     //     return {
+    //     //         proceed: true,
+    //     //         data: [event],
+    //     //         url: DATA_SCREENSHOT_ENDPOINT
+    //     //     }
+    //     // }
+    //
+    //     this.queue.push(event)
+    //     BStackLogger.debug(`Added data to request queue. Queue length = ${this.queue.length}`)
+    //
+    //     let data
+    //     const shouldProceed = this.shouldProceed()
+    //     if (shouldProceed) {
+    //         data = this.queue.splice(0, DATA_BATCH_SIZE)
+    //         this.resetEventBatchPolling()
+    //         BStackLogger.debug(`Sending data from request queue. Data length = ${data.length}, Queue length after removal = ${this.queue.length}`)
+    //     }
+    //
+    //     return {
+    //         proceed: shouldProceed,
+    //         data: data,
+    //         url: DATA_BATCH_ENDPOINT
+    //     }
+    // }
 
     add (event: UploadType) {
         if (!process.env.BS_TESTOPS_BUILD_COMPLETED) {
-            return {
-                proceed: false
-            }
-        }
-
-        if (!BATCH_EVENT_TYPES.includes(event.event_type)) {
-            return {
-                proceed: true
-            }
-        }
-
-        if (event.logs && event.logs[0] && event.logs[0].kind === 'TEST_SCREENSHOT') {
-            return {
-                proceed: true,
-                data: [event],
-                url: DATA_SCREENSHOT_ENDPOINT
-            }
+            throw new Error('Observability build start not completed yet.')
```

Comment:
is this handled ? We don't want to stop wdio execution for observability failure

Reply:
This is already existing code in master , didn't change anything over here